library/cpp/include/wavemap/pipeline/pipeline.h

```diff
@@ -26,43 +29,62 @@ class Pipeline {
         thread_pool_(thread_pool ? std::move(thread_pool)
                                  : std::make_shared<ThreadPool>()) {}
 
+  // Copy construction is not supported
+  Pipeline(const Pipeline&) = delete;
+
+  //! Deregister all measurement integrators and map operations
   void clear();
 
+  //! Returns true if an integrator with the given name has been registered
   bool hasIntegrator(const std::string& integrator_name) const;
+  //! Deregister the integrator with the given name. Returns true if it existed.
   bool eraseIntegrator(const std::string& integrator_name);
```

Comment:
Erase might be a bit misleading of a term as it could mean to fully delete it, as opposed to removing/deregistering it from the pipeline. I probably would use `removeIntegrator` given you use `addIntegrator` as well.

Reply:
When an integrator is added to the pipeline (by unique ptr move or by creating one for a given set of params), the pipeline takes over ownership. So `eraseIntegrator` does delete the integrator instance. Do you still prefer calling it `removeIntegrator` in this case? I could adjust the method s.t. it passes the integrator's unique ptr instance back to the caller as the return value. The caller could then cast it to bool to see if an instance existed, use it for something, or simply ignore it to have it automatically be deleted.